main.py

```diff
@@ -56,6 +57,29 @@ def is_uninteresting(tag: element.Tag):
     return tag.name in {"sup", "img"}
 
 
+def split_speeds(str) -> list:
+    braces = 0
+    res = []
+    current = []
```

Comment:
The string is built up character-by-character. At least in Java, every invocation of `current += char` creates a new string because String is immutable. So in Java, it is considered bad practice to build a string in a loop this way but instead a StringBuilder (basically an array of Strings) should be used: https://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java

Reply:
That may be the case in Java, and it may have been the case in Python in the distant past, but nowadays the performance difference is negligible, and the intent is much clearer. For illustration, I've built something similar in Python: https://gist.github.com/ianthetechie/3f9e27dcbae65ff14346c0e637e72f0b. The runtimes on my machine are almost exactly 30 sec for 1 million iterations for each strategy, with the join approach being slightly faster. I don't think that's worth optimizing for with the small amounts of data this script processes.

Reply:
Doesn't the linked script just do only 100 iterations? I also get about 30s each, but that would be incredibly slow, 333ms for one iteration, that can't be right?!